05.NablaTutorialExample/main.cpp

```diff
@@ -293,19 +293,21 @@ class NablaTutorialExampleApp : public ApplicationBase
 
 				We know ahead of time that `SBasicViewParameters` struct is the expected structure of the only UBO block in the descriptor set nr. 1 of the shader.
 			*/
-
-			IGPUBuffer::SCreationParams creationParams = {};
-			creationParams.canUpdateSubRange = true;
-			creationParams.usage = core::bitflag(asset::IBuffer::EUF_UNIFORM_BUFFER_BIT)|asset::IBuffer::EUF_TRANSFER_DST_BIT;
-			IDeviceMemoryBacked::SDeviceMemoryRequirements memReq;
-			memReq.vulkanReqs.size = sizeof(SBasicViewParameters);
-			memReq.vulkanReqs.alignment = physicalDevice->getLimits().UBOAlignment;
-			memReq.vulkanReqs.memoryTypeBits = 0xffffffffu;
-			memReq.memoryHeapLocation = IDeviceMemoryAllocation::ESMT_DEVICE_LOCAL;
-			memReq.mappingCapability = IDeviceMemoryAllocation::EMAF_NONE;
-			memReq.prefersDedicatedAllocation = true;
-			memReq.requiresDedicatedAllocation = true;
-			gpuubo = logicalDevice->createGPUBufferOnDedMem(creationParams,memReq);
+			{
+				IGPUBuffer::SCreationParams creationParams = {};
+				creationParams.canUpdateSubRange = true;
+				creationParams.usage = core::bitflag(asset::IBuffer::EUF_UNIFORM_BUFFER_BIT)|asset::IBuffer::EUF_TRANSFER_DST_BIT;
+				auto gpuubobuffer = logicalDevice->createBuffer(creationParams);
+
+				IDeviceMemoryBacked::SDeviceMemoryRequirements memReq;
+				memReq.prefersDedicatedAllocation = true;
+				memReq.requiresDedicatedAllocation = true;
+				//memReq.alignmentLog2 = physicalDevice->getLimits().UBOAlignment;
+				memReq.size = sizeof(SBasicViewParameters);
+				memReq.memoryTypeBits = physicalDevice->getDeviceLocalMemoryTypeBits();
```

Comment:
No @Hazardu,
That's not the way.
You first get the requirements from image/buffer you just created.
and then you do `&= physicalDevice->getXXXMemoryTypeBits();`
for example:
```
auto imageMemoryReqs = image->getMemoryReqs();
imageMemoryReqs.memoryTypeBits &= device->getPhysicalDevice()->getDeviceLocalMemoryTypeBits(); /
```
everything was explained here in this commit I made for you, https://github.com/Devsh-Graphics-Programming/Nabla/commit/6696500ae23873efbe2b89b85d64e1c647e57270